backend/plumber.go

```diff
@@ -125,18 +143,25 @@ func (i *Interface) addressAlreadyTaken() (bool, error) {
 	return false, nil
 }
 
+func (i *Interface) retryConnection(reason string) error {
+	log.Printf("Retry connect, reason: %s", reason)
+	time.Sleep(retryttl)
+	i.retries = i.retries + 1
+	if i.retries > maxretries-1 {
+		return fmt.Errorf("%s: Last error: %s", errMaxRetriesReached, reason)
+	}
+	return i.Connect()
```

Comment:
I think you should reset the counter here on success:

```go
 err := i.Connect()
if err == nil {
i.retries = 0
}
return err
```

That moves the retry state into this function and not spread throughout your program.  It is less brittle

Reply:
good point thanks, implemented this